tilequeue/queue/multisqs.py

```diff
@@ -0,0 +1,181 @@
+            msg_delay = 0
+            msg_tuple = (msg_id, coord_str, msg_delay)
+            msg_tuples.append(msg_tuple)
+            coord_ints.append(coord_int)
+
+        sqs_queue.write_batch(msg_tuples)
+        self.redis_client.sadd(self.inflight_key, *coord_ints)
+
+    def enqueue_batch(self, coords):
+        buf_per_queue = {}
+        n_queued = 0
+        n_in_flight = 0
+        for coord in coords:
+            # TODO log?
+            if not coord_is_valid(coord):
+                continue
+            coord_int = coord_marshall_int(coord)
+            if self._inflight(coord_int):
+                n_in_flight += 1
+            else:
+                n_queued += 1
+                sqs_queue_name = self.get_queue_name_for_zoom(coord.zoom)
+                if sqs_queue_name is None:
+                    # TODO log?
+                    continue
+                queue_buf = buf_per_queue.setdefault(sqs_queue_name, [])
+                queue_buf.append((coord, coord_int))
+                if len(queue_buf) == self.queue_buf_size:
+                    sqs_queue = self.sqs_queue_for_name.get(sqs_queue_name)
+                    assert sqs_queue_name, \
+                        'Missing queue for: %s' % sqs_queue_name
+                    self._write_batch(sqs_queue, queue_buf)
+                    del queue_buf[:]
+
+        for queue_name, queue_buf in buf_per_queue.items():
+            if queue_buf:
+                sqs_queue = self.sqs_queue_for_name.get(queue_name)
+                assert sqs_queue, 'Missing queue for: %s' % queue_name
+                self._write_batch(sqs_queue, queue_buf)
+
+        return n_queued, n_in_flight
+
+    def read(self, max_to_read=None):
+        if max_to_read is None:
+            max_to_read = self.queue_buf_size
+
+        coord_messages = []
+        left_to_read = max_to_read
+
+        for sqs_queue in self.sqs_queues:
+
+            queue_messages = sqs_queue.get_messages(
+                num_messages=left_to_read,
+                attributes=('SentTimestamp',))
+
+            for qm in queue_messages:
+
+                data = qm.get_body()
+                coord = deserialize_coord(data)
+                if coord is None:
+                    # TODO log?
+                    continue
+
+                try:
+                    timestamp = float(qm.attributes.get('SentTimestamp'))
+                except (TypeError, ValueError):
+                    timestamp = None
+
+                coord_message = CoordMessage(coord, qm, timestamp)
+                coord_messages.append(coord_message)
+                left_to_read -= 1
+
+            assert(left_to_read >= 0)
+            if left_to_read == 0:
+                break
+
+        return coord_messages
+
+    def job_done(self, coord_message):
+        queue_name = self.get_queue_name_for_zoom(coord_message.coord.zoom)
+        if queue_name is None:
+            # TODO log? should this assert instead?
```

Comment:
I think we should `assert` here. Presumably this would mean that tilequeue has rendered a zoom level for which it wasn't configured, which seems to me like an error condition. Should we assume that `get_queue_name_for_zoom` will return a valid queue name always, and have an assertion in that method (or use `[]` instead of `get()`)?